API.yaml

```diff
@@ -495,6 +497,10 @@ paths:
                 requested_items:
                 - 1
                 - 2
+                full_search:
+                  true/false
+                regex_map_filters:
+                  ["search_me"]
```

Comment:
General comment:
The TSP should be defined in the why it makes sense from the API point of view, independent of the back-end implementation. The front-end and TSP are agnostic to the back-end implementation and definitions in the TSP needs to independent as well. Having said that, we can get inspired by the Trace Compass trace server reference implementation. The names of variables, data structures don't need to match what is done in the Trace Compass back-end.

I'm wondering if the timegraph filter should be an object that has option ;full_search' and the reqex_map_filter in it.

Secondly, there needs to be an description on what is the data provided in the `regex_map_filter` and what is the expected outcome (i.e. back-end reply) when such a filter is applied. Reading the TSP alone, a client or server developer would not know what to pass and what to do with the reply.

Reply:
I suggest an optional filter data structure `FilterQueryParameters` which has an optional `isFullSearch` and a required `filterExpressionsMap`. 

In the TSP, this data structure `FilterQueryParameters` will be added with key `filter_query_paramters`.

Reply:
Hi,

Can you see my new PR?